File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int> prev_nums;
        int max_len=0;
        for(int i=0;i<nums.size();i++)
        {
            prev_nums[nums[i]]=1;
        }
        int curr=0;
        int curr_idx=0;
        int curr_len=0;
        for(int i=0;i<nums.size();i++)
        {
            if(prev_nums.count(nums[i]-1))
            {
                continue;
            }
            curr = nums[i];
            while(prev_nums.count(curr))
            {
                curr_len++;
                max_len=max(curr_len,max_len);
                curr++;
            }
            curr_len=0;
        }
        return max_len;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        max_len = 0
        max_l = 0
        max_r = 0
        for i in range(n):
            l = i
            r = i
            while l>=0 and r<n:
                if s[l] == s[r]:
                    if (r-l+1)>max_len:
                        max_len = max(max_len,(r-l+1))
                        max_l = l
                        max_r=r
                    l-=1
                    r+=1
                else:
                    break
            
            l=i
            r=i+1
            while l>=0 and r<n:
                if s[l] == s[r]:
                    if (r-l+1)>max_len:
                        max_len = max(max_len,(r-l+1))
                        max_l = l
                        max_r= r
                    l-=1
                    r+=1
                else:
                    break
            
        return s[max_l:max_r+1]

            
